File: Examples/VisualBasic/Tasks/GetOutlookTaskDetails.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic

Namespace Tasks
    Public Class GetOutlookTaskDetails
        Public Shared Sub Run()
            ' ExStart:GetOutlookTaskDetails
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tasks()

            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("Aspose.one"))

            ' Get all RichText nodes
            Dim nodes As IList(Of RichText) = oneFile.GetChildNodes(Of RichText)()

            ' Iterate through each node
            For Each richText As RichText In nodes
                For Each tag In richText.Tags
                    If TypeOf tag Is NoteTask Then
                        Dim noteTask As NoteTask = DirectCast(tag, NoteTask)
                        ' Retrieve properties
                        Console.WriteLine("Completed Time: " + noteTask.CompletedTime.ToString())
                        Console.WriteLine("Create Time: " + noteTask.CreationTime.ToString())
                        Console.WriteLine("Due Date: " + noteTask.DueDate.ToString())
                        Console.WriteLine("Status: " + noteTask.Status.ToString())
                        Console.WriteLine("Task Type: " + noteTask.TaskType.ToString())
                        Console.WriteLine("Icon: " + noteTask.Icon.ToString())
                    End If
                Next
            Next
            ' ExEnd:GetOutlookTaskDetails
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Tasks/GetOutlookTaskDetails.cs

using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.Tasks
{
    using System.Linq;

    public class GetOutlookTaskDetails
    {
        public static void Run()
        {
            // ExStart:GetOutlookTaskDetails
            // ExFor:RichText.Tags
            // ExFor:ITag
            // ExFor:ITag.CompletedTime
            // ExFor:ITag.CreationTime
            // ExFor:ITag.Icon
            // ExFor:ITag.Status
            // ExFor:NoteTask
            // ExFor:NoteTask.DueDate
            // ExSummary:Shows how to access details of outlook's tasks.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tasks();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "Aspose.one");

            // Get all RichText nodes
            IList<RichText> nodes = oneFile.GetChildNodes<RichText>();

            // Iterate through each node
            foreach (RichText richText in nodes)
            {
                var tasks = richText.Tags.OfType<NoteTask>();
                if (tasks.Any())
                {
                    Console.WriteLine($"Task: {richText.Text}");
                    foreach (var noteTask in tasks)
                    {
                        // Retrieve properties
                        Console.WriteLine($"    Completed Time: {noteTask.CompletedTime}");
                        Console.WriteLine($"    Create Time: {noteTask.CreationTime}");
                        Console.WriteLine($"    Due Date: {noteTask.DueDate}");
                        Console.WriteLine($"    Status: {noteTask.Status}");
                        Console.WriteLine($"    Icon: {noteTask.Icon}");
                    }
                }
            }

            // ExEnd:GetOutlookTaskDetails
        }
    }
}